osl/sandbox.go

```diff
@@ -166,6 +166,11 @@ type Interface interface {
 	// and moving it out of the sandbox.
 	Remove() error
 
+	// SetSecondaryIP adds or deletes the passed IP as a secondary IP on
+	// the interface. for ex: set the vip of services in the same network
+	// as secondary IP.
+	SetSecondaryIP(ip *net.IPNet, add bool) error
```

Comment:
`this interface which only has getter methods`
We do have Remove() method in this interface. Also, so far we haven't had a requirement to dynamically modify an osl interface.

If we add a method like `UpdateInterface` to the `Sandbox` interface it has to be generic; ie: the option setter should encapsulate the parameters for a given update operation. There shouldn't be extra arguments that are specific to one particular operation. What can be done though is to have two different option setters, like `addalias` and `removealias`. I don't know if this approach has any particular advantage though. Lets see if we can get a consensus on this.